build/scripts/build-utils.ps1

```diff
@@ -110,9 +110,66 @@ function Exec-Script([string]$script, [string]$scriptArgs = "") {
 # Ensure that NuGet is installed and return the path to the 
 # executable to use.
 function Ensure-NuGet() {
-    Exec-Block { & (Join-Path $PSScriptRoot "download-nuget.ps1") } | Out-Host
-    $nuget = Join-Path $repoDir "NuGet.exe"
-    return $nuget
+    $nugetVersion = "4.1.0"
+    $toolsDir = Join-Path $binariesDir "Tools"
+    Create-Directory $toolsDir
+
+    $destFile = Join-Path $toolsDir "NuGet.exe"
+    $versionFile = Join-Path $toolsDir "NuGet.exe.version"
+
+    # Check and see if we already have a NuGet.exe which exists and is the correct
+    # version.
+    if ((Test-Path $destFile) -and (Test-Path $versionFile)) {
+        $scratchVersion = Get-Content $versionFile
+        if ($scratchVersion -eq $nugetVersion) {
+            return $destFile
+        }
+    }
+
+    Write-Host "Downloading NuGet.exe"
+    $webClient = New-Object -TypeName "System.Net.WebClient"
+    $webClient.DownloadFile("https://dist.nuget.org/win-x86-commandline/v$nugetVersion/NuGet.exe", $destFile)
+    $nugetVersion | Out-File $versionFile
+    return $destFile
+}
+
+# Checks to see if a particular version of the SDK is available on %PATH%. This is 
+# how MSBuild locates the SDK. 
+function Test-SdkInPath([string]$version) {
+    foreach ($part in ${env:PATH}.Split(';')) {
+        $dotnetExe = Join-Path $part "dotnet.exe"
+        if (Test-Path $dotnetExe) {
+            $sdkPath = Join-Path $part "sdk"
+            $sdkPath = Join-Path $sdkPath $version
+            return Test-Path $sdkPath
+        }
+    }
+
+    return $false
+}
+
+# Ensure the proper SDK in installed in our %PATH%. This is how MSBuild locates the 
+# SDK.
+function Ensure-SdkInPath() { 
+    $sdkVersion = "2.0.0-preview3-006923"
```

Comment:
I think it would be useful to include all versions in a single file.

See e.g. https://github.com/dotnet/symreader/blob/master/build/Versions.props#L9, https://github.com/dotnet/symreader/blob/master/build/Versions.props#L15.

It's very easy to extract such version from both powershell script and, indeed, use it in msbuild props and targets files as well.